pyctcdecode/decoder.py

```diff
@@ -62,17 +63,70 @@
 # store frame information for each word, where frame is the logit index of (start_frame, end_frame)
 Frames = Tuple[int, int]
 WordFrames = Tuple[str, Frames]
-# all the beam information we need to keep track of during decoding
-# text, next_word, partial_word, last_char, text_frames, part_frames, logit_score
-Beam = Tuple[str, str, str, Optional[str], List[Frames], Frames, float]
-# same as BEAMS but with current lm score that will be discarded again after sorting
-LMBeam = Tuple[str, str, str, Optional[str], List[Frames], Frames, float, float]
-
-# for output beams we return the text, the scores, the lm state and the word frame indices
-# text, last_lm_state, text_frames, logit_score, lm_score
-OutputBeam = Tuple[str, Optional[AbstractLMState], List[WordFrames], float, float]
-# for multiprocessing we need to remove kenlm state since it can't be pickled
-OutputBeamMPSafe = Tuple[str, List[WordFrames], float, float]
+
+
+@dataclasses.dataclass(frozen=True)
+class Beam:
+    """Contains all the info needed for decoding a beam."""
+
+    text: str
+    next_word: str
+    partial_word: str
+    last_char: Optional[str]
+    text_frames: List[Frames]
+    partial_frames: Frames
+    logit_score: float
+
+    @classmethod
+    def from_lm_beam(cls, lm_beam: LMBeam) -> "Beam":
+        """Create a Beam from an LMBeam by stripping out the LM score."""
+        return Beam(
+            text=lm_beam.text,
+            next_word=lm_beam.next_word,
+            partial_word=lm_beam.partial_word,
+            last_char=lm_beam.last_char,
+            text_frames=lm_beam.text_frames,
+            partial_frames=lm_beam.partial_frames,
+            logit_score=lm_beam.logit_score,
+        )
+
+
+@dataclasses.dataclass(frozen=True)
+class LMBeam:
+    """Contains everything in a beam plus an additional language model score."""
```

Comment:
Can this be made a subclass of `Beam` that just adds the LM score? I've never done that with a dataclass, so I'm not even sure if that would work.